### PR TITLE
Try to start after cluster recovery

### PR DESCRIPTION
**Pull request by tomyouyou**
If it fails, the broker will hang here forever.

### CODE REVIEW

**Comment by lukebakken** [comment]
Hello! Could you let us know how this was discovered?

File: deps/rabbit/src/rabbit_node_monitor.erl [hunk 1]
```diff
@@ -769,7 +769,13 @@ do_run_outside_app_fun(Fun) ->
 wait_for_cluster_recovery(Condition) ->
     ping_all(),
     case Condition() of
-        true  -> rabbit:start();
+        true  -> 
+            rabbit_log:info("to start after cluster recovery"),
+            try
+                rabbit:start()
+            catch _:_ ->
```
Comment (lukebakken) [inline deps/rabbit/src/rabbit_node_monitor.erl:776]: Any caught exception here should be logged.

**Review by lukebakken — CHANGES_REQUESTED** [review]

**Comment by tomyouyou** [comment]
> Hello! Could you let us know how this was discovered?
@ lukebakken 
In July, in my cluster environment, there was a network failure and a node could not be recovered. The broker has gone to the recovery startup process, but it hung up after an exception. Later, it was recovered by manually restarting the rabbitmq service.
But the log file at that time could not be found again.

**Comment by michaelklishin** [comment]
We don't have enough information to work with and this stops the node with the exit code of `0` on any exception. This is a curious approach and doesn't seem to be necessary if the goal is to get logs (which I think you will anyway, from the crash report).

So it's not clear how to reproduce or what problem exactly we are solving => we cannot accept this.

**Pull request closed without merging**